beeline/middleware/awslambda/__init__.py

```diff
@@ -0,0 +1,99 @@
+import beeline
+
+# In Lambda, a cold start is when Lambda has to spin up a new instance of a
+# function to satisfy a request, rather than re-use an existing instance.
+# This usually has a non-trivial effect on latency for the request and is
+# worth instrumenting.
+COLD_START = True
+
+def _get_trace_payload(trace_string):
+    # the first value is the trace payload version
+    # at this time there is only one version, so there's no need to do anything
+    # based on this
+    _, data = trace_string.split(';', 1)
+
+    kv_pairs = data.split(',')
+
+    trace_id, parent_id, context = None, None, None
+    # For version 1, we expect three kv pairs. If there's anything else, the
+    # payload is malformed and we do nothing.
```

Comment:
Node and Go allow for extra unknown keys eg `1;trace_id=aoeu,parent_id=snth,context=aoeu,unknown_key=snth` is still valid; the unknown keys are ignored.